crates/shielded_token/src/masp/shielded_wallet.rs

```diff
@@ -437,7 +437,7 @@ pub trait ShieldedApi<U: ShieldedUtils + MaybeSend + MaybeSync>:
     async fn precompute_asset_types<C: Client + Sync>(
         &mut self,
         client: &C,
-        tokens: Vec<&Address>,
+        tokens: BTreeSet<&Address>,
```

Comment:
this could just be an `impl Iterator<Item=&Address>`

Reply:
The motivation for this change was to try and avoid doing the same precomputation twice (with the redundant querying and encoding that comes with it) if the input contains duplicates, not that this is actually a serious problem.